### PR TITLE
Replace with kate

### PR DESCRIPTION
**Pull request by NoCtrlZ**
## Problem
The VerifyFailure::ConstraintNotSatisfied struct and we need to indicate cell_values in that field.
It’s a Vec<(VirtualCell, String)> but does any one know how to get it?

### CODE REVIEW

File: keccak256/Cargo.toml
Comment (CPerezz) [inline keccak256/Cargo.toml:10]: Shouldn't we try to merge your `halo2` changes into `appliedzkp/halo2` and use it from there?

Also, bear in mind that the patch in the workspace `Cargo.toml` will overwrite this info.

File: keccak256/src/gates/rho.rs
Comment (CPerezz) [inline keccak256/src/gates/rho.rs:262]: Why has this been commented out?? Not sure I follow TBH. 

I think it has to do with the comment above. We should merge your halo2 work first into our fork and then, use that fork here as dependency. 

Then you'll not need to comment out things like this.

File: zkevm-circuits/Cargo.toml
Comment (CPerezz) [inline zkevm-circuits/Cargo.toml:11]: ditto

File: zkevm-circuits/src/gadget/evm_word.rs
Comment (CPerezz) [inline zkevm-circuits/src/gadget/evm_word.rs:174]: If we're not using this anymore, we should remove it.

File: zkevm-circuits/src/gadget/evm_word.rs
Comment (CPerezz) [inline zkevm-circuits/src/gadget/evm_word.rs:283]: Why is the test commented out?

File: zkevm-circuits/src/gadget/evm_word.rs
Comment (CPerezz) [inline zkevm-circuits/src/gadget/evm_word.rs:291]: ditto

**Review by CPerezz — CHANGES_REQUESTED** [review]
Some tests have been commented out without any specific reason or comment.
Also, there's an entire bench that has also been commented out without any specific reason or comment.

Aside from that, I think we should not rely on @NoCtrlZ fork and instead we should make the changes that Shin has there into `appliedzkp/halo2` and use that as a dependency. I belive that will allow yo uncomment the tests.

Finally I would like to add that It would be really helpful if you could add some commit messages or descriptions in the PR for some of the decisions taken like the removal/commenting some tests. As otherways is impossible to know why it has been done and it's also much easier to miss it. And I consider it important when it comes to testing removal.

Aside from that, nice work! :smile:

File: keccak256/Cargo.toml
Comment (NoCtrlZ) [reply keccak256/Cargo.toml:10]: This is the PR of this but it depends other PR so it takes time to be merged.
https://github.com/appliedzkp/halo2/pull/8
Should I wait for these PR merged?

File: keccak256/src/gates/rho.rs
Comment (NoCtrlZ) [reply keccak256/src/gates/rho.rs]: It seems the API was changed but I fixed.
I will comment from the next time.

**Comment by NoCtrlZ** [comment]
Hi @CPerezz 
> Finally I would like to add that It would be really helpful if you could add some commit messages or descriptions in the PR for some of the decisions taken like the removal/commenting some tests.

Where should I mention this one?
I already mentioned about `evm_word` test failure with previous PR and discord channel.
Should I describe like `Todo:` in code?

**Comment by CPerezz** [comment]
> Hi @CPerezz
> 
> > Finally I would like to add that It would be really helpful if you could add some commit messages or descriptions in the PR for some of the decisions taken like the removal/commenting some tests.
> 
> Where should I mention this one? I already mentioned about `evm_word` test failure with previous PR and discord channel. Should I describe like `Todo:` in code?

Either in the commit message or in the PR description in the top of the page which is now empty.
If the test fails due to the lookup fn inclusion and hasn't been solved yet I'm not sure if we should then merge it yet. WDYT @han0110 ??
If we want to do it, then just add an `#[ignore]` in the top of the test instead of commenting it out.

Also, let's wait to see what @han0110 and @kilic think about merging your `halo2` fork into `appliedzkp` one and then use that one as the dependency no? Or are there any news that I'm missing on that regard? Since I remember, https://github.com/appliedzkp/halo2/pull/8 should be ready to be merged. And therefore we can simply use the `appliedzkp` fork of `halo2` directly.

Finally, which Discord you refer? You mean TG?

**Comment by NoCtrlZ** [comment]
> If we want to do it, then just add an #[ignore] in the top of the test instead of commenting it out.

Copy that!
> Since I remember, appliedzkp/halo2#8 should be ready to be merged

My PR is depending on this https://github.com/appliedzkp/halo2/pull/1.
It's already approved by all reviewers so it should be merged before mine.

> Finally, which Discord you refer? You mean TG?

Yes telegram.

File: zkevm-circuits/src/evm_circuit.rs
Comment (han0110) [inline zkevm-circuits/src/evm_circuit.rs:168]: This should not be renamed
```suggestion
    FixedLookup(FixedLookup, [Expression<F>; 3]),
    BusMappingLookup(BusMappingLookup<F>),
```

**Review by han0110 — COMMENTED** [review]
Some enum and function name with suffix `lookup` seem to be replaced to `lookup_any`, which I think it unnecessary. Are they accidentally replaced?

**Comment by han0110** [comment]
The test of `evm_word` fails here, the hashed result is larger than the field:

https://github.com/appliedzkp/zkevm-circuits/blob/4300f363a00961374297bf1f265032ac6d745026/zkevm-circuits/src/gadget/evm_word.rs#L32-L33

It should be fixed by:

```diff
-    let r: [u8; 32] = hasher.fixed_result().as_slice().try_into().unwrap(); 
-    F::from_bytes(&r).unwrap()
+    let mut r = [0; 64];
+    r[..32].copy_from_slice(hasher.fixed_result().as_slice());
+    F::from_bytes_wide(&r)
```

**Comment by NoCtrlZ** [comment]
I fixed `evm_test` and lookup mischanges.

@han0110 
> The test of evm_word fails here, the hashed result is larger than the field:
btw, how did you realize this one?
I debugged and realized but it took so much time than you.
I would appreciate you could tell me how to debug?

Thank you.

**Review by CPerezz — APPROVED** [review]
LGTM modulo the `halo2` dep update to `appliedzkp` upstream and a review approval from @han0110

**Review by han0110 — APPROVED** [review]
LGTM!

> btw, how did you realize this one?

running test with `RUST_BACKTRACE=1` would show where it panics, it should tell more informations.

File: keccak256/src/gates/iota_b13.rs
Comment (CPerezz) [inline keccak256/src/gates/iota_b13.rs:118]: Why have these selectors been forced to use `complex_selector` now?

Comment (NoCtrlZ) [reply keccak256/src/gates/iota_b13.rs:118]: `zcash` decided not to selector in lookup.
https://github.com/zcash/halo2/commit/145d096dcd6f840178e97ec2bfd6ca9c6201093e#diff-c7d3f048cff35b9298e74eb82f24b7755a72927c699f387552c0ac97f7c6c4a8R1224

**Pull request merged**
